torch/csrc/jit/script/compiler.cpp

```diff
@@ -1683,6 +1683,45 @@ struct to_ir {
     }
   }
 
+  void emitAugAssignmentGeneric(
+      const AugAssign& stmt,
+      const Subscript& lhs,
+      Value* sliceable) {
+    // Get the idx to augment
+    const auto subscriptExprs = lhs.subscript_exprs();
+    const TypePtr type = sliceable->type();
+    TypePtr elemType = nullptr;
+
+    if (const ListTypePtr listType = type->cast<ListType>()) {
+      elemType = listType->getElementType();
+    } else if (const DictTypePtr dictType = type->cast<DictType>()) {
+      elemType = dictType->getKeyType();
+    }
+
+    if (subscriptExprs.size() != 1 || elemType == nullptr) {
```

Comment:
These are actually two different errors. The lhs is "Slices are not supported". The rhs is "this type doesn't support augmented assignment at all". Can we differentiate between these two?